enterprise/views.py

```diff
@@ -916,11 +933,17 @@ def post(self, request, enterprise_uuid, course_id):
             # the learner to course specific DSC with enterprise UUID from
             # there the learner will be directed to the ecommerce flow after
             # providing DSC.
+            query_string_params = {
+                'course_mode': selected_course_mode_name,
+            }
+            if enterprise_catalog_uuid:
+                query_string_params.update({'enterprise_customer_catalog_uuid': enterprise_catalog_uuid})
+
             next_url = '{handle_consent_enrollment_url}?{query_string}'.format(
```

Comment:
I see we're passing the enterprise_customer_catalog_uuid to the `handle_consent_enrollment_url` 👍 but I'm missing where that is then preserving the catalog_uuid when it redirects to the ecomm flow - are we sure we're handling that case?

Reply:
@yes. That case is handled. Since the `enterprise_customer_catalog_uuid` is included in the `next_url` so it in case of successful consent the view `PayAndVerifyView` in [LMS will this value](https://github.com/edx/edx-platform/pull/18470/) and then it will be passed over to Ecommerce basket page just like the `sku` (`http://localhost:8002/basket/add/?sku=F97B5FF&enterprise_customer_catalog_uuid=6eca3efb-f3a0-4c08-806f-c6e6b65d61cb`) and then the basket will apply the related enterprise catalog conditional offer.